.github/workflows/workflow.yml

```diff
@@ -0,0 +1,29 @@
+name: Run unit tests
+on: ["push", "pull_request"]
+
+jobs:
+  test:
+    name: Dub tests
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        dc: [dmd-latest, ldc-latest]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: 'Install D compiler'
+      uses: dlang-community/setup-dlang@v1
+      with:
+        compiler: ${{ matrix.dc }}
+
+    - name: 'Build and test'
+      run: |
+        # Test multiple builds
+          dub test --build=debug
+          dub test --build=release
+          dub test --build=release-debug
+          dub test --build=release-inline
+        # Coverage
+          dub test --coverage
```

Comment:
You should run `--coverage` on every test and pass `--DRT-covopt="merge:1"` to merge them. See https://dlang.org/articles/code_coverage.html . This way everything is ensure to be covered by conditional compilation flags.

For coverage to be practical, add some coverage service like codecov. See https://github.com/codecov/codecov-action#usage .

Reply:
Damn, too many flags :sweat:. I'll fix it in a minute. Coverage service will be another PR.

Reply:
> Damn, too many flags . I'll fix it in a minute. Coverage service will be another PR.

Note that the last flag to merge coverage is a runtime flag, so you need to pass it as the last argument like this `-- --DRT-covopt="merge:1"`